### PR TITLE
KOGITO-9372: Move --delete to undeploy command

### PR DESCRIPTION
**Pull request by ederign**
On this PR, based on feedback, we are moving ./cli deploy --delete. to ./cli undeploy

Artifacts: https://drive.google.com/drive/u/1/folders/1Nhm1NHfOGT3RoldD9gl8ZoP0s-z0pt8f

### CODE REVIEW

File: packages/kn-plugin-workflow/pkg/command/deploy_undeploy_common.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
+package command
```
Comment (caponetto) [inline packages/kn-plugin-workflow/pkg/command/deploy_undeploy_common.go:17]: Please add the copyright header to this file.

File: packages/kn-plugin-workflow/pkg/command/deploy_undeploy_common.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+import (
+	"fmt"
+	"github.com/kiegroup/kie-tools/packages/kn-plugin-workflow/pkg/common"
+	"github.com/kiegroup/kie-tools/packages/kn-plugin-workflow/pkg/metadata"
+	"github.com/kiegroup/kogito-serverless-operator/workflowproj"
+	"os"
+	"path/filepath"
+)
+
+type DeployUndeployCmdConfig struct {
```
Comment (caponetto) [inline packages/kn-plugin-workflow/pkg/command/deploy_undeploy_common.go:28]: Maybe name this as `DeploymentCmdConfig` ?

File: packages/kn-plugin-workflow/pkg/command/undeploy.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```
Comment (caponetto) [inline packages/kn-plugin-workflow/pkg/command/undeploy.go]:
```suggestion
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
```

File: packages/kn-plugin-workflow/pkg/command/undeploy.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	if err = deleteDeploy(&cfg); err != nil {
+		return fmt.Errorf("❌ ERROR: undeploying: %w", err)
+	}
+
+	fmt.Println("\n🎉 Kogito Serverless Workflow project successfully undeployed.")
+
+	return nil
+}
+
+func deleteDeploy(cfg *DeployUndeployCmdConfig) error {
```
Comment (caponetto) [inline packages/kn-plugin-workflow/pkg/command/undeploy.go]:
```suggestion
func undeploy(cfg *DeployUndeployCmdConfig) error {
```
A suggestion just to keep the new pattern.

**Review by caponetto — COMMENTED** [review]
Minor comments inline :)

File: packages/kn-plugin-workflow/pkg/command/deploy_undeploy_common.go (hunk 2 above)
Comment (ederign) [reply packages/kn-plugin-workflow/pkg/command/deploy_undeploy_common.go:28]: This is used both for deploy/Undeploy (common from both)./

File: packages/kn-plugin-workflow/pkg/command/deploy_undeploy_common.go (hunk 1 above)
Comment (ederign) [reply packages/kn-plugin-workflow/pkg/command/deploy_undeploy_common.go:17]: fixed!

File: packages/kn-plugin-workflow/pkg/command/undeploy.go (hunk 3 above)
Comment (ederign) [reply packages/kn-plugin-workflow/pkg/command/undeploy.go]: fixed!

File: packages/kn-plugin-workflow/pkg/command/undeploy.go (hunk 4 above)
Comment (ederign) [reply packages/kn-plugin-workflow/pkg/command/undeploy.go]: fixed!

**Review by caponetto — APPROVED** [review]

**Review by zdrapela — APPROVED** [review]
Looking good! Thanks

**Pull request merged**